components/HamburgerMenu/index.js

```diff
@@ -0,0 +1,116 @@
+import styled from "styled-components";
+import { stack as Menu } from "react-burger-menu";
+import { Icon } from "@/components/Icon";
+import { useRouter } from "next/router";
+import { StyledLink } from "@/components/StyledLinks";
+
+export default function HamburgerMenu({
+  menuOpen,
+  setMenuOpen,
+  getRandomActivity,
+}) {
+  const router = useRouter();
+
+  function handleStateChange(state) {
+    setMenuOpen(state.isOpen);
+  }
+
+  return (
+    <StyledMenu
+      width={"25%"}
+      right
+      isOpen={menuOpen}
+      onStateChange={handleStateChange}
+      disableOverlayClick={true}
+      customBurgerIcon={false}
+      customCrossIcon={false}
+      noOverlay
+    >
+      <StyledLink href="/">
+        <StyledNavIcon active={router.pathname === "/"}>
```

Comment:
You should prefix props you pass to a styled-component with "$". Otherwise, you will get an error message in your console.
And naming this prop "isActive" would be even better, since that would directly indicate to other developers that it is a `boolean`.

```suggestion
        <StyledNavIcon $isActive={router.pathname === "/"}>
```